### PR TITLE
EventPreFilters added

### PR DESCRIPTION
**Pull request by vlisovsk**
Proposed solution for #17 .

### CODE REVIEW

File: first-analysis-steps/minimal-dv-job.md [hunk 1]
```diff
@@ -133,6 +133,22 @@ to the `UserAlgorithms` list.
 The `TupleFile` attribute defines the name of the ROOT output file that DaVinci
 will store any algorithm output in, which should be our ntuple.
 
+{% callout "Being smart and efficient" %}
+`DaVinci` offers a possibility to loop only over events passing a certain requirement, rather than all events in the DST. This feature can save a lot of processing time and computing resources when running over millions of events.
```
Comment (alexpearce) [inline first-analysis-steps/minimal-dv-job.md]: * "a possibility" -> "the possibility"

Is it worth explaining why can save a lot of time?

File: first-analysis-steps/minimal-dv-job.md [hunk 2]
```diff
@@ -133,6 +133,22 @@ to the `UserAlgorithms` list.
...
+For example, one may require events passing a specific stripping or trigger line requirement:
+```python
+from PhysConf.Filters import LoKi_Filters
+fltrs = LoKi_Filters (
+    STRIP_Code = """
+    HLT_PASS_RE('StrippingD2hhPromptDst2D2KKLineDecision')
+    """
+    )
```
Comment (alexpearce) [inline first-analysis-steps/minimal-dv-job.md]: Maybe put the string on one line.

```python
fltrs = LoKi_Filters (
    STRIP_Code="HLT_PASS_RE('StrippingD2hhPromptDst2D2KKLineDecision')"
)
```

File: first-analysis-steps/minimal-dv-job.md [hunk 3]
```diff
@@ -133,6 +133,22 @@ to the `UserAlgorithms` list.
...
+DaVinci().EventPreFilters = fltrs.filters('Filters')
+```
+Here we use the [LoKi functor `HLT_PASS_RE`](http://lhcb-doxygen.web.cern.ch/lhcb-doxygen/davinci/latest/d7/dae/namespace_lo_ki_1_1_cuts.html#aee4bba9ae8443acd970dd52e20e5b8c1) which is a simple predicate to check the positive decision on (in this case) the stripping line.
```
Comment (alexpearce) [inline first-analysis-steps/minimal-dv-job.md]: "which is a simple predicate to check the" -> "which checks for a"

File: first-analysis-steps/minimal-dv-job.md (hunk 1 above)
Comment (alexpearce) [reply first-analysis-steps/minimal-dv-job.md]: Also: might be worth explicitly stating that using filters is _strongly_ recommended, to save yourself time and to save our collaboration's resources.

Comment (vlisovsk) [reply first-analysis-steps/minimal-dv-job.md]: Could you please precise what do you mean by 'Is it worth explaining why can save a lot of time?'?  You wish to add some arguments more precise than simply 'looping over events passing the requirement'?

[the remark this answers is not in the document]
Comment (alexpearce) [reply first-analysis-steps/minimal-dv-job.md]: @vlisovsk I meant the same thing as @saschastahl here. The reason why prefilters can save so much time is that they don't need to run the unpacking. The unpacking is pretty much-all-or-nothing, but you generally only care about the existence of a single TES location. Instead, you can check the Stripping and/or trigger decisions, where a positive decision makes it much more likely (or certain) that the TES location(s) you want exist.

But, this is quite a techinical detail, and we don't explain the packing anywhere else, so talking about it here is probably just confusing.

[the remark this answers is not in the document]
Comment (vlisovsk) [reply first-analysis-steps/minimal-dv-job.md]: Well, thank you for improving my understanding of the topic 👍 
Will try to write this all in a short and understandable way now.

**Comment by vlisovsk** [comment]
Any comments on the updated version? I am not sure if this 'simplified' explanation is correct and sufficient.

**Comment by alexpearce** [comment]
A really excellent addition, thank you! 🍨

**Comment by saschastahl** [comment]
Indeed, thanks a lot for that!

**Pull request merged**